polaris-react/src/components/LegacyCard/components/Header/Header.tsx

```diff
@@ -32,12 +33,23 @@ export function Header({children, title, actions}: LegacyCardHeaderProps) {
   );
 
   const headingMarkup =
+    // eslint-disable-next-line no-nested-ternary
     actionMarkup || children ? (
-      <LegacyStack alignment="baseline">
-        <LegacyStack.Item fill>{titleMarkup}</LegacyStack.Item>
-        {actionMarkup}
-        {children}
-      </LegacyStack>
+      polarisSummerEditions2023 ? (
+        <HorizontalStack wrap={false} align="space-between" blockAlign="center">
```

Comment:
I think this might need a gap here so really long title don't butt up against the actions?